src/IRCNetwork.cpp

```diff
@@ -942,6 +942,35 @@ bool CIRCNetwork::DelChan(const CString& sName) {
     return false;
 }
 
+bool CIRCNetwork::MoveChan(const CString& sName, unsigned int uIndex) {
+    if (uIndex >= m_vChans.size()) return false;
+
+    auto it = m_vChans.begin();
+    for (; it != m_vChans.end(); ++it)
```

Comment:
Initially I thought this would require `std::find_if` with a rather ugly lambda. I see now that passing a `CChan*` from `CIRCNetwork::FindChan` would work.

Reply:
I don't see how this would be an improvement. Also `std::find_if` would be needed.